src/cluster/src/start/k8.rs

```diff
@@ -49,103 +51,27 @@ static MAX_SC_NETWORK_LOOP: Lazy<u16> = Lazy::new(|| {
-    /// The docker registry to use
-    image_registry: String,
-    /// The name of the Fluvio helm chart to install
-    chart_name: String,
-    /// A specific version of the Fluvio helm chart to install
-    chart_version: String,
-    /// The location to find the fluvio charts
-    chart_location: ChartLocation,
-    /// The name of the SPU group to create
-    group_name: String,
-    /// The name of the Fluvio cloud
-    cloud: String,
-    /// Whether to save an update to the Fluvio profile
-    save_profile: bool,
-    /// Whether to install fluvio-sys with fluvio-app
-    install_sys: bool,
-    /// Whether to update the `kubectl` context
-    update_context: bool,
-    /// Whether to upgrade an existing installation
-    upgrade: bool,
-    /// How much storage to allocate on SPUs
-    spu_spec: SpuGroupSpec,
-    /// The logging settings to set in the cluster
-    rust_log: Option<String>,
-    /// The TLS policy for the SC and SPU servers
-    server_tls_policy: TlsPolicy,
-    /// The TLS policy for the client
-    client_tls_policy: TlsPolicy,
-    /// The authorization ConfigMap name
-    authorization_config_map: Option<String>,
-    /// Should the pre install checks be skipped
-    skip_checks: bool,
-    /// if set use cluster ip instead of egress
-    use_cluster_ip: bool,
-    /// if set, disable spu liveness checking
-    skip_spu_liveness_check: bool,
-    /// chart values
-    chart_values: Vec<PathBuf>,
-    /// Whether to print check results as they are performed
-    render_checks: bool,
-}
-
-impl ClusterInstallerBuilder {
-    /// Creates a [`ClusterInstaller`] with the current configuration.
-    ///
-    /// This may fail if there is a problem conencting to Kubernetes or
-    /// finding the `helm` executable on the local system.
-    ///
-    /// # Example
-    ///
-    /// The simplest flow to create a `ClusterInstaller` looks like the
-    /// following:
-    ///
-    /// ```no_run
-    /// # use fluvio_cluster::ClusterInstaller;
-    /// let installer = ClusterInstaller::new()
-    ///     .build()
-    ///     .expect("should create ClusterInstaller");
-    /// ```
-    ///
-    /// [`ClusterInstaller`]: ./struct.ClusterInstaller.html
-    pub fn build(self) -> Result<ClusterInstaller, ClusterError> {
-        Ok(ClusterInstaller {
-            config: self,
-            kube_client: K8Client::default().map_err(K8InstallError::K8ClientError)?,
-            helm_client: HelmClient::new().map_err(K8InstallError::HelmError)?,
-        })
-    }
-
+#[derive(Builder, Debug)]
+#[builder(build_fn(skip), setter(prefix = "with"))]
+pub struct ClusterConfig {
     /// Sets the Kubernetes namespace to install Fluvio into.
     ///
     /// The default namespace is "default".
     ///
     /// # Example
     ///
-    /// ```no_run
-    /// # use fluvio_cluster::ClusterInstaller;
-    /// let installer = ClusterInstaller::new()
+    /// ```
+    /// # use fluvio_cluster::{ClusterConfig, ClusterConfigBuilder, ClusterError};
```

Comment:
Suggest simplifying doc.   Instead of duplicating examples.  Just have a single example.   Then each setting will just detail what it does.

Reply:
The reason for having individual docs is so that the doc tests will fail if we change the interface. I actually want to open another issue because it turns out our CI has not actually been running doc tests and our docs were out of date.

Reply:
Still like to have a just a single example